zerver/models/users.py

```diff
@@ -877,6 +877,9 @@ def can_create_web_public_streams(self) -> bool:
             return False
         return self.has_permission("can_create_web_public_channel_group")
 
+    def can_manage_default_streams(self) -> bool:
+        return self.is_realm_admin
+
```

Comment:
Probably we should make sure the main endpoint for changing default streams uses this too.

Reply:
Opened #32577